From f7895bd654a9f98be958eda99b5a95496e6743d2 Mon Sep 17 00:00:00 2001
From: Huy Hoang <boydaihungst@gmail.com>
Date: Sat, 28 Oct 2023 02:42:44 +0700
Subject: [PATCH] Every-voldmanaged-storage-is-adoptable

---
 main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.cpp b/main.cpp
index e299476..d19f051 100644
--- a/main.cpp
+++ b/main.cpp
@@ -275,7 +275,7 @@ static int process_config(VolumeManager* vm, VoldConfigs* configs) {
             int partnum = entry.partnum;
             int flags = 0;
 
-            if (entry.is_encryptable()) {
+            if (entry.is_encryptable() || true) {
                 flags |= android::vold::Disk::Flags::kAdoptable;
                 configs->has_adoptable = true;
             }
-- 
2.42.0

