From ecc8c3fa39fc1d3ffeab997d9fc75dfc0a74a89d Mon Sep 17 00:00:00 2001
From: Huy Hoang <boydaihungst@gmail.com>
Date: Mon, 30 Oct 2023 20:53:30 +0700
Subject: [PATCH] Mini vendor/evolution

---
 config/common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/common.mk b/config/common.mk
index 4856f2a6..9bfb4ffa 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -205,7 +205,7 @@ TARGET_IS_PIXEL_TABLET ?= false
 TARGET_PIXEL_STAND_SUPPORTED ?= false
 TARGET_SUPPORTS_QUICK_TAP ?= false
 TARGET_USES_LEGACY_BOOTANIMATION ?= false
-TARGET_USES_MINI_GAPPS ?= false
+TARGET_USES_MINI_GAPPS ?= true
 TARGET_USES_PICO_GAPPS ?= false
 
 # Face Unlock
-- 
2.42.0

