From 6db7a65894a3b9ec91d74233e07e8c36b2409796 Mon Sep 17 00:00:00 2001
From: Huy Hoang <boydaihungst@gmail.com>
Date: Sat, 28 Oct 2023 02:41:05 +0700
Subject: [PATCH] Ignore-dm-verity-setup-issues

---
 init/first_stage_mount.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/init/first_stage_mount.cpp b/init/first_stage_mount.cpp
index 8ef20ecf9..506f3d1e8 100644
--- a/init/first_stage_mount.cpp
+++ b/init/first_stage_mount.cpp
@@ -440,7 +440,6 @@ bool FirstStageMount::MountPartition(const Fstab::iterator& begin, bool erase_sa
     if (begin->fs_mgr_flags.avb) {
         if (!SetUpDmVerity(&(*begin))) {
             PLOG(ERROR) << "Failed to setup verity for '" << begin->mount_point << "'";
-            return false;
         }
     } else {
         LOG(INFO) << "AVB is not enabled, skip verity setup for '" << begin->mount_point << "'";
-- 
2.42.0

